libtiledbsoma/src/utils/arrow_adapter.cc

```diff
@@ -377,6 +383,8 @@ std::string_view ArrowAdapter::to_arrow_format(
     tiledb_datatype_t datatype, bool use_large) {
     switch (datatype) {
         case TILEDB_STRING_ASCII:
+            return use_large ? "Z" : "z";  // large because TileDB
```

Comment:
* @nguyenv noted she has a fix (https://github.com/single-cell-data/TileDB-SOMA/pull/2305#issuecomment-2015509453) but that is large and has CI fails
* This PR is my attempt to extract the essentials to unblock 1.9 for issue #2306 
* When I have `"Z"` here then the repro script in the description field of this PR passes. But then other existing unit-test cases fail, since what should read back as `["a","b"]` reads back as `[b"a", b"a"]`
* Yet when I have `"U"` here, then the repro script in the description field of this PR fails -- the reported problem is simply not fixed -- but all existing unit-test cases pass.

I seem to recall that `STRING_ASCII` _along with_ some other flag, metadata something-something, is used to differentiate between string-as-bytes and string-as-string ... @nguyenv / @ihnorton perhaps you recall how TileDB-Py does this (as it successfully does, in the code snippet included in the description field of this PR).

Reply:
https://github.com/single-cell-data/TileDB-SOMA/compare/main...viviannguyen/array-write-path#diff-c66edd736f9a9cb059c6f6e0f4cc6d1cdfa0fd91a2b9220dd2e83c12c2546089R451-R457
https://github.com/single-cell-data/TileDB-SOMA/compare/main...viviannguyen/array-write-path#diff-c66edd736f9a9cb059c6f6e0f4cc6d1cdfa0fd91a2b9220dd2e83c12c2546089R407

I believe these are the snippets that should fix the issue you're seeing.

Reply:
I specially handle the string types for enums like that rather relying on `to_tiledb_format`.

Reply:
@nguyenv both of the links you sent me open up with "57 files changed" -- non-navigable

Nonetheless I'll take a look at what you wrote in Slack 🤞